library/manageiq_provider.py

```diff
@@ -305,9 +307,9 @@ def main():
     provider_type = module.params['type']
     state         = module.params['state']
     zone          = module.params['zone']
-    hostname      = module.params['hostname']
-    port          = module.params['port']
-    token         = module.params['token']
+    hostname      = module.params['provider_api_hostname']
+    port          = module.params['provider_api_port']
+    token         = module.params['auth_token']
```

Comment:
Is the `auth_token` here to auth to the provider?

Assuming so, I think it should probably follow the same pattern as the other two and be cause `provider_api_auth_token`. It's a bit long, but I don't mind that.